pykafka/cluster.py

```diff
@@ -287,4 +288,8 @@ def update(self):
                         'will NOT work. You need to create at least one topic '
                         'manually using the Kafka CLI tools.')
         self._update_brokers(metadata.brokers)
-        self._update_topics(metadata.topics)
+        try:
+            self._update_topics(metadata.topics)
+        except LeaderNotAvailable:
```

Comment:
Actually alternatively, shouldn't we retry `Cluster.update` altogether, when this is caught? I mean, we just got sent `metadata` which seems to already be out of date while we're trying to use it.